src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+            throw new DataProcessingException("Cannot insert book into DB", e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();;
+            if (resultSet.next()) {
+                Book book = getBookFromResultSet(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get book with ID: " + id + " from DB", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                books.add(getBookFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get all books from DB", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books "
+                + "SET title = ?, price = ?"
+                + "WHERE id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot update book with ID: "
+                    + book.getId() + " in DB", e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            int rowsAffected = preparedStatement.executeUpdate();
+            return rowsAffected > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot delete book with ID: " + id + " from DB", e);
+        }
+    }
+
+    @Override
+    public boolean clearTable() {
+        String query = "TRUNCATE TABLE books;";
```

Comment:
Use uppercase for SQL keywords to follow the SQL style best practices. For example, use 'TRUNCATE TABLE' instead of 'truncate table'.